### PR TITLE
Fcma io

### PR DESCRIPTION
**Pull request by yidawang**
refactor the IO code in fcma examples to `brainiak/fcma` to address #158, and modify `.gitignore` and `setup.py` accordingly.

Thanks for reviewing!

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,8 +6,10 @@
 *.lib
 *.o
 *.npy
+!tests/fcma/data/*.npy
 *.log
 *.nii.gz
+!tests/fcma/data/*.nii.gz
```
Comment (mihaic) [inline .gitignore]: No need for this. Instead, add the files with `git add -f`.

File: .gitignore [hunk 2]
```diff
@@ -29,4 +31,4 @@ __MACOSX
 
 brainiak/fcma/cython_blas.c
 brainiak/eventseg/_utils.c
-brainiak/examples/fcma/face_scene/
+examples/fcma/face_scene/
```
Comment (mihaic) [inline .gitignore]: I can see a lot of lines like this showing up over time. Please remove it and keep your datasets outside of BrainIAK if you prefer them not to show up in `git status`.

File: brainiak/fcma/io.py [hunk 3]
```diff
@@ -24,46 +31,49 @@
...
+    """ read data in NIfTI from a dir and apply the spatial mask to them
 
     Parameters
     ----------
     dir: str
         the path to all subject files
     file_extension: str
         the file extension, usually nii.gz or nii
-    mask_file: str
+    mask_file: str, default None
```
Comment (mihaic) [inline brainiak/fcma/io.py]: You should document this as `Optional[str]`:
https://www.python.org/dev/peps/pep-0484/#union-types

File: .gitignore (hunk 1 above)
Comment (yidawang) [inline .gitignore]: done

File: brainiak/fcma/io.py (hunk 3 above)
Comment (yidawang) [inline brainiak/fcma/io.py]: done

**Comment by mihaic** [comment]
Jenkins, retest this please.

**Comment by mihaic** [comment]
@TuKo, do you have any idea what is happening with the SSSRM tests?
```
__________________________________ test_sssrm __________________________________

    def test_sssrm():
        import os
        os.environ['THEANO_FLAGS'] = 'device=cpu, floatX=float64'
    
        from sklearn.utils.validation import NotFittedError
        import numpy as np
        import brainiak.funcalign.sssrm
    
        voxels = 100
        align_samples = 400
        samples = 500
        subjects = 2
        features = 3
        n_labels = 4
    
        model = brainiak.funcalign.sssrm.SSSRM(n_iter=5, features=features, gamma=10.0, alpha=0.1)
        assert model, "Invalid SSSRM instance!"
    
        # Create a Shared response S with K = 3
        theta = np.linspace(-4 * np.pi, 4 * np.pi, samples)
        z = np.linspace(-2, 2, samples)
        r = z**2 + 1
        x = r * np.sin(theta)
        y = r * np.cos(theta)
    
        S = np.vstack((x, y, z))
        S_align = S[:, :align_samples]
        S_classify = S[:, align_samples:]
        X = []
        Z = []
        Z2 = []
        y = []
        Q, R = np.linalg.qr(np.random.random((voxels, features)))
        X.append(Q.dot(S_align) + 0.1 * np.random.random((voxels, align_samples)))
        Z.append(Q.dot(S_classify) + 0.1 * np.random.random((voxels, samples - align_samples)))
        Z2.append(Q.dot(S_classify) + 0.1 * np.random.random((voxels, samples - align_samples)))
        y.append(np.repeat(np.arange(n_labels), (samples - align_samples)/n_labels))
    
        # Create more subjects align and classification data
        for subject in range(1, subjects):
            Q, R = np.linalg.qr(np.random.random((voxels, features)))
            X.append(Q.dot(S_align) + 0.1 * np.random.random((voxels, align_samples)))
            Z2.append(Q.dot(S_classify) + 0.1 * np.random.random((voxels, samples - align_samples)))
    
        # Create more subjects labels data
        for subject in range(1, subjects):
            y.append(np.repeat(np.arange(n_labels), (samples - align_samples)/n_labels))
    
        # Set the logging level to INFO
        import logging
        logging.basicConfig(level=logging.INFO)
    
        # Check that runs with 2 subject
>       model.fit(X, y, Z2)

tests/funcalign/test_sssrm.py:169: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
brainiak/funcalign/sssrm.py:195: in fit
    self.w_, self.s_, self.theta_, self.bias_ = self._sssrm(X, Z, new_y)
brainiak/funcalign/sssrm.py:346: in _sssrm
    w = self._update_w(data_align, data_sup, labels, w, s, theta, bias)
brainiak/funcalign/sssrm.py:524: in _update_w
    minstep = np.min((10**-np.floor(np.log10(f_subject))), 1e-1)
../brainiak_pr_venv_KGNDz/lib/python3.5/site-packages/numpy/core/fromnumeric.py:2349: in amin
    return amin(axis=axis, out=out, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

a = array(1.0), axis = 0.1, out = None, keepdims = False

    def _amin(a, axis=None, out=None, keepdims=False):
>       return umr_minimum(a, axis, None, out, keepdims)
E       TypeError: 'float' object cannot be interpreted as an integer

../brainiak_pr_venv_KGNDz/lib/python3.5/site-packages/numpy/core/_methods.py:29: TypeError
----------------------------- Captured stderr call -----------------------------
INFO:brainiak.funcalign.sssrm:Starting SS-SRM
INFO:brainiak.funcalign.sssrm:Objective function 4.136741
INFO:brainiak.funcalign.sssrm:Iteration 1
INFO:brainiak.funcalign.sssrm:Subject Wi 0
```

**Comment by TuKo** [comment]
@mihaic no Idea. It is crashing in numpy np.min() function.

**Comment by mihaic** [comment]
@TuKo, it looks to me like you are passing `1e-1` as `axis` to `np.min`:
`minstep = np.min((10**-np.floor(np.log10(f_subject))), 1e-1)`
https://github.com/IntelPNI/brainiak/blob/master/brainiak/funcalign/sssrm.py#L524

**Comment by TuKo** [comment]
That is what is happening now. It is supposed to do minimum between two scalar values.
Did we change some library version for testing? Why is this failing now (after several PRs since september)?

**Comment by yidawang** [comment]
Nothing changed as you can tell from the history...

**Comment by TuKo** [comment]
Well, you added nibabel as a dependency. So there are environment changes.
Not sure if this is related to the failing cause, though.

**Comment by yidawang** [comment]
Two commits before, `nibabel` was also included and nothing went wrong then...

**Pull request merged**